app/routes/user/password.server.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import { ChangePasswordCommand } from '@aws-sdk/client-cognito-identity-provider'
+
+import { getOpenIDClient, storage } from '../__auth/auth.server'
+import { getUser, parseTokenSet } from '../__auth/user.server'
+import { client, maybeThrow } from './cognito.server'
+
+export class PasswordManager {
+  #oldPassword: string
+  #newPassword: string
+  #accessToken: string
+
+  private constructor(
+    oldPassword: string,
+    newPassword: string,
+    accessToken: string
+  ) {
+    this.#oldPassword = oldPassword
+    this.#newPassword = newPassword
+    this.#accessToken = accessToken
+  }
+
+  static async create(
+    request: Request,
+    oldPassword: string,
+    newPassword: string
+  ) {
+    const user = await getUser(request)
+    if (!user) throw new Response('not signed in', { status: 403 })
+    const client = await getOpenIDClient()
+    const session = await storage.getSession(request.headers.get('Cookie'))
+    const refreshToken = session.get('refreshToken')
+    const tokenSet = await client.refresh(refreshToken)
+    const parsedTokenSet = parseTokenSet(tokenSet)
+
```

Comment:
We are currently not doing any validation for password length or strength etc. what are the acceptable parameters for the password that you would like to use if any?